gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1268 @@
+            })
+
+        # start with the most significant core.
+        sorted_clusters = sorted(sorted_clusters,
+            key=lambda cluster: (
+                cluster["amount_parent_labels"],
+                cluster["label"]  # makes sorting deterministic
+            ), reverse=True)
+
+        def remove_from_all_sets(label):
+            """removes a label from every set in "parent_labels" for
+            each core in sorted_clusters."""
+            for a in sorted_clusters:
+                if label in a["parent_labels"]:
+                    a["parent_labels"].remove(label)
+
+        # APPLYING THE RULES 1 to 3
+        # iterate over the cluster labels, see which clusters/labels
+        # are valid to cause the creation of a stable topic
+        for i, cluster in enumerate(sorted_clusters):
+            label = cluster["label"]
+            # label is iterating over 0, 1, 3, ...
+
+            # 1. rule - remove if the cores in the cluster have no parents
+            if cluster["amount_parent_labels"] == 0:
+                # label is NOT VALID
+                cluster["is_valid"] = False
+                remove_from_all_sets(label)
+
+            # 2. rule - remove if it has less than min_cores as parents
+            # (parents are always also cores)
+            if len(cluster["parent_labels"]) < min_cores:
+                cluster["is_valid"] = False
+                remove_from_all_sets(label)
+
+            # 3. checking for "easy_valid"s
+            # checks if the core has at least min_cores of cores with the only label as itself
+            if cluster["amount_parent_labels"] >= 1:
+                if sum(map(lambda x: x == {label}, cluster["parent_labels"])) >= min_cores:
+                    cluster["is_valid"] = True
+
+        # Reaplying the rule 3
+        # this happens when we have a close relationship among 2 or more clusters
+        for cluster in [cluster for cluster in sorted_clusters if np.isnan(cluster["is_valid"])]:
+
+            # this checks for parent_labels, which are also modified in this function
+            # hence order will influence the result. it starts with the most significant
+            # label (hence "sorted_clusters")
+            if sum(map(lambda x: x == {label}, cluster["parent_labels"])) >= min_cores:
+                cluster["is_valid"] = True
+            else:
+                cluster["is_valid"] = False
+                # removes a label from every set in parent_labels for each core
+                for a in sorted_clusters:
+                    if label in a["parent_labels"]:
+                        a["parent_labels"].remove(label)
+
+        # list of all the label numbers that are valid
+        valid_labels = np.array([cluster["label"] for cluster in sorted_clusters if cluster["is_valid"]])
+
+        for topic in results:
+            topic["valid_parents"] = {label for label in topic["parent_labels"] if label in valid_labels}
+
+        def validate_core(core):
```

Comment:
Can this be a module-scope _internal function? If yes, then move it out. The outer method is already far too long.